subsys/net/lib/nrf_cloud/src/nrf_cloud.c

```diff
@@ -220,14 +285,175 @@ void nrf_cloud_process(void)
 	nct_process();
 }
 
+#if IS_ENABLED(CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD)
+static int start_connection_poll()
+{
+	if (current_state == STATE_IDLE) {
```

Comment:
Could the ```NOT_VALID_STATE``` macro be used here? Out of scope of this PR but would be nice to handle states with common function calls instead of handling the state variables directly.